tests/tests/helpers.rs

```diff
@@ -133,19 +146,30 @@ pub async fn server_with_subscription() -> SocketAddr {
 }
 
 pub async fn server() -> SocketAddr {
-	let server = ServerBuilder::default().build("127.0.0.1:0").await.unwrap();
-	let mut module = RpcModule::new(());
-	module.register_method("say_hello", |_, _| "hello").unwrap();
+	#[derive(Debug, Clone)]
+	struct ConnectionDetails<S> {
+		inner: S,
+		connection_id: u32,
+	}
 
-	module
-		.register_async_method_with_details(
-			"raw_method",
-			|_, connection_details, _| async move { connection_details.id() },
-		)
-		.unwrap();
+	impl<'a, S> RpcServiceT<'a> for ConnectionDetails<S>
+	where
+		S: RpcServiceT<'a>,
+	{
+		type Future = S::Future;
+
+		fn call(&self, mut request: jsonrpsee::types::Request<'a>) -> Self::Future {
+			request.extensions_mut().insert(self.connection_id);
+			self.inner.call(request)
+		}
+	}
+
+	let mut module = RpcModule::new(());
+	module.register_method("say_hello", |_, _, _| "hello").unwrap();
+	module.register_method("raw_method", |_, _, ext| *ext.get::<u32>().unwrap()).unwrap();
```

Comment:
Nit: I wonder whether we should rename this to eg "get_connection_id" to make it clearer what it's doing/testing?